### ISSUE TITLE
Mouse tracking does not respect the system's `rows to scroll` setting

### ISSUE THREAD

**Issue by o-sdn-o** [microsoft/terminal#18102]
### Windows Terminal version

current main

### Windows build number

10.0.19045.4894

### Other Software

WSL + vim

### Steps to reproduce

Step to reproduce:
1. Set the `rows to scroll` system setting to 5.
2. Run
  ```
  wsl bash -c "vim -c :h"
  ```
3. Scroll the mouse wheel.

Type `:q` and press `Enter` (repeat twice) to quit.

### Expected Behavior

The cursor should move at the pace specified in the system-wide settings - 5 rows per 1 wheel step.

https://github.com/user-attachments/assets/81ccadb8-454c-44a9-bff6-9a046c0ab1d8

### Actual Behavior

The cursor moves at a speed of one row per mouse wheel step.

https://github.com/user-attachments/assets/aad547ef-c99d-4b32-9002-b35f26091c63

**Comment by zadjii-msft** [comment]
I'm surprised that this wasn't already somewhere on our backlog! I could have swore I remembered a discussion about this already, but I sure can't find it. Thanks for filing!

**Comment by lhecker** [comment]
I thought that when mouse tracking is enabled, the terminal only sends a single sequence per scroll event. Isn't sending of multiple cursor up/down sequences only a thing during the alternate scroll mode with mouse tracking disabled?

**Comment by DHowett** [comment]
Yes, this should be the case. Xterm mouse reporting reports *button events,* of which each wheel detent is one.

Xterm mouse reporting does not report lines scrolled.

How many lines the application scrolls when it receives a wheel detent button reports an application choice.

We can choose to treat ASB with Alternate Scroll differently, but I am not interested in changing how wheel detents are reported.

**Comment by zadjii-msft** [comment]
I assumed that this was alternate scroll mode in action, fwiw.

**Comment by DHowett** [comment]
Hm. I expect vim to use xterm mouse reporting.

It does here:

![Image](https://github.com/user-attachments/assets/980bd4be-c7bf-46b5-b839-8f9f4bd3ded6)

**Comment by o-sdn-o** [comment]
There is no way for an application inside WSL to know the system-wide scrolling settings. Moreover, for example in Midnight Commander the scrolling speed is hardcoded as 2 lines per wheel step for the viewer and editor, and 1 line everywhere else. Repeating mouse scroll reports in the amount of the system-wide scrolling setting value makes such applications "live" and acceptable/comfortable when using scrolling.

The only place where I see a discrepancy is Far Manager - on their side they multiply the scrolling by the system-wide scrolling setting value, so they can reach it.

**Comment by DHowett** [comment]
> the scrolling speed is hardcoded as 2 lines per wheel step

Well that's not good! Terminal cannot know that, so if it sends 3 (or 5!) wheel down events, it'll scroll 6 (or 10!) lines. I rather think this is an argument _not_ to do what you are proposing.

**Comment by o-sdn-o** [comment]
This is very much the case with touchpads/mice with precise scrolling, as with them it is possible to scroll one line at a time if necessary, despite the fact that the system-wide setting is greater than one.

**Comment by o-sdn-o** [comment]
Here is an example of touchpad scrolling with `rows to scroll=10` in the Midnight Commander viewer running inside a vtm that multiplies mouse reports.

![Image](https://github.com/user-attachments/assets/3d142d7c-d393-430e-98e7-7a74bbd729cf)

https://github.com/user-attachments/assets/e9b63cdb-51e5-4f3f-b5cb-9889f215a4a0
